pandas/tests/indexes/test_category.py

```diff
@@ -598,17 +598,32 @@ def test_reindex_dtype(self):
         tm.assert_index_equal(res, exp, exact=True)
         tm.assert_numpy_array_equal(indexer, np.array([0, 3, 2], dtype=np.intp))
 
-    def test_reindex_duplicate_target(self):
+    def test_reindex_duplicate_source(self):
         # See GH23963
         c = CategoricalIndex(["a", "b", "c", "a"], categories=["a", "b", "c", "d"])
-        with pytest.raises(ValueError, match="non-unique indexer"):
-            c.reindex(["a", "a", "c"])
+        with pytest.raises(ValueError, match="duplicate axis"):
+            c._can_reindex(["a", "c"])
```

Comment:
don't test internal methods, rather the user facing

Reply:
I will delete this test entirely then, because `_can_reindex` is called directly by the `BlockManager`, not any of the index machinery itself. `test_reindexing` in `tests/indexing/test_categorical.py` should still cover the relevant code.